src/content/reference/react/use.md

```diff
@@ -0,0 +1,769 @@
+
+A component that passes a promise to `use` integrates with [`Suspense`](/reference/react/Suspense) and [`ErrorBoundary`](/reference/react/Component#catching-rendering-errors-with-an-error-boundary). When a pending promise is passed to `use`, the fallback of the closest Suspense Boundary will be displayed. When the promise is resolved, the component calling `use` will be rendered using the data from the resolved promise. The rendered component will then replace the Suspense Boundary fallback. If the promise passed to `use` is rejected, the fallback of the nearest Error Boundary will be displayed. 
+
+[See more examples below.](#usage)
+
+#### Parameters {/*parameters*/}
+
+* `resource`: this is the source of the data you want to read a value from. A resources can be a [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) or a [context](/learn/passing-data-deeply-with-context).
+
+#### Returns {/*returns*/}
+
+The `use` Hook returns the value that was read from the resource like the resolved value of a [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) or [context](/learn/passing-data-deeply-with-context).
+
+#### Caveats {/*caveats*/}
+
+* The parent function of the `use` Hook must be a React component or your own Hooks.
+* `use` is not recommended for data fetching from a [Server Component](/reference/react/components#server-components). `async` and `await` are preferred when fetching data from a Server Component.
+* When passing promises to `use`, you must store the promise in a state variable or context to avoid recreating the promise on every rerender. When invoking a function that returns a promise, you can to store the returned promise with a cache mechanism like React's `cache` function.
+
+---
+
+## Usage {/*usage*/}
+
+### Reading promises with `use` {/*reading-promises-with-use*/}
+
+`use` functions similarly to the `await` with JavaScript promises but is compatible with React's renderer and [`Suspense`](/reference/react/Suspense). Browser APIs or data fetching libraries that return a promise can be passed to `use` to read the value of the promise.
+
+In the example below we'll use <CodeStep step={1}>[`navigator.storage.estimate()`](https://developer.mozilla.org/en-US/docs/Web/API/StorageManager/estimate)</CodeStep>, a browser API that provides information on how much storage is being used by the current page and how much space is available. <CodeStep step={1}>`navigator.storage.estimate()`</CodeStep> is an async function that returns a Promise which resolves once the information is available.
+
+```js [[1,6, "navigator.storage.estimate()"], [2,6, "storagePromise"], [3,6, "useState"], [4,7, "use"], [5,8, "storage.quota"]]
+"use client";
+
+import { use, useState } from "react"
+
+function StorageInfo() {
+  const [storagePromise] = useState(navigator.storage.estimate());
```

Comment:
Is this usage correct @acdlite? I thought at present the promise needs to be created before render.

Reply:
The example does work in `18.3.0-canary-9377e1010-20230712` ([see this](https://codesandbox.io/s/dwvpmr)) which is the current canary version Next.js uses. But if its not supposed to work/isn't something we want to support let me know and I'll fix this.

Reply:
What I'm unclear on is the guarantees for when state is preserved when a component suspends before mounting. If this works then I suppose useMemo also would, although since navigator.storage.estimate() isn't pure I suppose it's technically wrong to call it from useMemo.

Reply:
per @acdlite creating promises during rendering is not a supported pattern. In the future it might be a good idea to write docs for how to work with promises when writing a data fetching library that integrates with Suspense. I'll be rewriting the examples to avoid creating promises as part of rendering.

Reply:
I've rewritten this page to reflect that promises can't be created as a part of rendering.